tools/lib/NQP/Config/NQP.pm

```diff
@@ -100,13 +100,67 @@ sub configure_misc {
 sub configure_moar_backend {
     my $self  = shift;
     my $imoar = $self->{impls}{moar};
+    my $config = $self->{config};
 
     $self->gen_moar;
 
+    my $moar_config = $self->moar_config;
+
     $self->{config}{moar} = $imoar->{config}{moar};
 
     $self->sorry( "No moar executable found in " . $self->cfg('prefix') )
       unless $self->is_executable( $self->{config}{moar} );
+
+    if ( $config->{relocatable} eq 'reloc' ) {
+        $config->{static_nqp_home}          = '';
+        $config->{static_nqp_home_define}   = '';
+    }
+    else {
+        my $qchar = $config->{quote};
+        $config->{static_nqp_home} =
+          File::Spec->rel2abs(
+            File::Spec->catdir( $config->{'prefix'}, 'share', 'nqp' )
+          );
+        $config->{static_nqp_home_define} =
+          '-DSTATIC_NQP_HOME='
+          . $qchar . $self->c_escape_string( $config->{static_nqp_home} ) . $qchar;
+    }
+
+    # Strip rpath from ldflags so we can set it differently ourself.
+    $config->{ldflags} = $moar_config->{'moar::ldflags'};
```

Comment:
All those config variables are strictly Moar-related. Therefore they belong to `$self->{impls}{moar}` (`$imoar` shortcut in this routine). That, actually, pointed me out to the fact that same change must be done in rakudo where `configure_moar_backend` sets them directly on the main context too.

The good reason for this: if we ever get another backend with supported compilation of runners it would use different set of flags.

Reply:
I moved them to the moar config hash. I did the same in the rakudo PR.